.github/workflows/build-binaries-and-push-to-r2.yaml

```diff
@@ -0,0 +1,46 @@
+name: Upload core binary to Cloudflare R2 bucket
+
+on:
+  push:
+    branches:
+      - '*'
+
+jobs:
+    build-and-upload:
+        name: Build binary from current commit and push to R2
+        runs-on: ubuntu-22.04
+        permissions:
+            contents: write
+            packages: write
+            id-token: write
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+              with:
+                fetch-depth: "0"
+                path: axelar-core
+                submodules: recursive
+            
+            - name: Build axelard binary with commit hash
+              id: build-binary
+              run: |
+                mkdir ./binary_file/
+                cd axelar-core
+                make build
+                commit_hash=$(git rev-parse --short HEAD)
+                echo "The commit hash: $commit_hash"
+                cd ..
+                sudo mv axelar-core/bin/axelard ./binary_file/axelard_$commit_hash
+                echo "binary-directory=./binary_file" >> $GITHUB_OUTPUT
```

Comment:
follow the naming convention of the binary's internal versioning and our mainnet releases.

1. the version is setup in the Makefile using `git describe --tags` which includes the first 7 characters of commit hash. so should use that in the name.
2.  `./bin/axelard-"$OS"-"$ARCH"-"$SEMVER"` is the naming pattern. should follow this. instead of SEMVER you would have output from axelard version or you can directory use `git describe --tags`

Reply:
changed to this:
```
semver=$(git describe --tags)
sudo mv axelar-core/bin/axelard ./binary_file/axelard-linux-amd64-$semver
```